File: app/src/main/kotlin/com/app/backpackr/BackPackRApp.java
package com.app.backpackr;

import android.app.Application;

import com.app.backpackr.api.ApiBaseUrls;
import com.app.backpackr.dagger.components.DaggerNetworkComponent;
import com.app.backpackr.dagger.components.NetworkComponent;
import com.app.backpackr.dagger.modules.AppModule;
import com.app.backpackr.dagger.modules.NetworkModule;

import javax.inject.Inject;

/**
 * Created by kmikhailovskiy on 25.11.2016.
 */

public class BackPackRApp extends Application {
    @Inject
    NetworkComponent networkComponent;

    @Override
    public void onCreate() {
        super.onCreate();
        networkComponent = DaggerNetworkComponent.builder()
                .appModule(new AppModule(this))
                .networkModule(new NetworkModule(ApiBaseUrls.GOOGLE_PLACES.toString()))
                .build();
    }
}

File: app/src/main/kotlin/com/app/backpackr/BackPackRApp.kt
package com.app.backpackr

import android.app.Application
import com.app.backpackr.injection.components.AppComponent
import com.app.backpackr.injection.components.DaggerAppComponent
import com.app.backpackr.injection.components.DaggerDatabaseComponent
import com.app.backpackr.injection.components.DatabaseComponent
import com.app.backpackr.injection.modules.AppModule
import io.realm.Realm

/**
 * Created by kmikhailovskiy on 01.12.2016.
 */

class BackPackRApp : Application() {
    companion object {
        lateinit var appComponent: AppComponent
        lateinit var databaseComponent: DatabaseComponent
    }

    override fun onCreate() {
        super.onCreate()
        Realm.init(this)
        appComponent = DaggerAppComponent.builder()
                .appModule(AppModule(this))
                .build()
        databaseComponent = DaggerDatabaseComponent.builder()
                .build()
    }
}